jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/ByteAccumulator.java

```diff
@@ -19,56 +19,92 @@
 package org.eclipse.jetty.websocket.common.extensions.compress;
 
 import java.nio.ByteBuffer;
-import java.util.ArrayList;
-import java.util.List;
 
+import org.eclipse.jetty.io.ByteBufferAccumulator;
+import org.eclipse.jetty.io.ByteBufferPool;
 import org.eclipse.jetty.util.BufferUtil;
 import org.eclipse.jetty.websocket.api.MessageTooLargeException;
 
-public class ByteAccumulator
+/**
+ * @deprecated use {@link ByteBufferAccumulator} instead.
+ */
+@Deprecated
+public class ByteAccumulator implements AutoCloseable
 {
-    private final List<byte[]> chunks = new ArrayList<>();
+    private static final int MIN_SPACE = 8;
+
+    private final ByteBufferAccumulator accumulator;
     private final int maxSize;
     private int length = 0;
 
     public ByteAccumulator(int maxOverallBufferSize)
+    {
+        this(maxOverallBufferSize, null);
+    }
+
+    public ByteAccumulator(int maxOverallBufferSize, ByteBufferPool byteBufferPool)
     {
         this.maxSize = maxOverallBufferSize;
+        this.accumulator = new ByteBufferAccumulator(byteBufferPool);
     }
 
-    public void copyChunk(byte[] buf, int offset, int length)
+    public int getLength()
     {
-        if (this.length + length > maxSize)
+        return accumulator.getLength();
```

Comment:
you have both a length field and getLength method on the accumulator.  Use one or the other.

Reply:
Yep, should be using `length` here.

Note that the `getLength()` on the `ByteBufferAccumulator` is inefficient, it adds up the remaining on all buffers each time. So thats why we are using this `length` field.

Reply:
check overflow as well here